sdk/state/close.go

```diff
@@ -89,18 +116,27 @@ func (c *Channel) ProposeClose() (CloseAgreement, error) {
 	c.latestUnauthorizedCloseAgreement = CloseAgreement{
 		Details:               d,
 		CloseSignatures:       txClose.Signatures(),
-		DeclarationSignatures: append([]xdr.DecoratedSignature{}, c.latestAuthorizedCloseAgreement.DeclarationSignatures...),
+		DeclarationSignatures: txDecl.Signatures(),
 	}
 	return c.latestUnauthorizedCloseAgreement, nil
 }
 
 func (c *Channel) validateClose(ca CloseAgreement) error {
-	latestWithoutObservation := c.latestAuthorizedCloseAgreement.Details
-	latestWithoutObservation.ObservationPeriodTime = 0
-	latestWithoutObservation.ObservationPeriodLedgerGap = 0
-
-	if ca.Details != latestWithoutObservation {
-		return fmt.Errorf("close agreement details do not match saved latest authorized close agreement")
+	if ca.Details.IterationNumber != c.latestAuthorizedCloseAgreement.Details.IterationNumber {
```

Comment:
nvrm (after looking at payment.go again) since we can compare `ConfirmingSigner` sgtm to leave on `CloseAgreementDetails`

Reply:
We need it on the details because it forms part of the agreement and part of the transactions. But that's a good point that they are not easily comparable. If we stored keys as strings throughout the SDK then we wouldn't have this issue. Also, if we stored a non-pointer that'd also solve the problem. If we did either of those it'd be good to do it consistently for all keys in the SDK, so let's figure that out separate to this.